src/k8s/pkg/snap/util/node.go

```diff
@@ -0,0 +1,30 @@
+package snaputil
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils"
+)
+
+func IsWorker(s snap.Snap) (bool, error) {
```

Comment:
General question: Do we have something like a guideline when to use a util function and when to put it into the `Snap` interface?
e.g. for this function it would also make sense to have it in the interface. Otherwise you'd need to always create a lock file in the tests at `s.WorkerNodeLockFile`. If we have `Snap.IsWorker` we can just mock the return value.

Reply:
Summoning @neoaggelos for the discussion as we've talked about this. Afaik, the snap interface should just hold paths and configs and be simple while functional bits should go in the `snaputils`.

Reply:
Well, this is what I thought but we also got e.g. `StartService` there.

Reply:
yy, a ServiceManager of sorts should grow from there, but that's something we can/should do at a later time.